### PR TITLE
remove EnqueueRequestForObject

### PR DESCRIPTION
**Pull request by mingmin-ddex**
refactored code, separate pvc, pv, and storage-class reconciliation into different controllers

they all reside in kalm_pvc_controller.go before this

ref: https://quip.com/yyouAxGhkS1N/Kalm-Controller-#fRcACA352V8

### CODE REVIEW

File: controller/controllers/kalm_ns_controller.go [hunk 1]
```diff
@@ -52,14 +53,22 @@ func NewKalmNSReconciler(mgr ctrl.Manager) *KalmNSReconciler {
...
+
 func (r *KalmNSReconciler) SetupWithManager(mgr ctrl.Manager) error {
 	return ctrl.NewControllerManagedBy(mgr).
 		For(&v1.Namespace{}).
-		//For(&v1alpha1.HttpsCertIssuer{}).
-		//For(&v1alpha1.HttpsCert{}).
-		Watches(&source.Kind{Type: &v1alpha1.HttpsCertIssuer{}}, &handler.EnqueueRequestForObject{}).
-		Watches(&source.Kind{Type: &v1alpha1.HttpsCert{}}, &handler.EnqueueRequestForObject{}).
-		//Owns(&v1alpha1.HttpsCert{}).
+		Watches(&source.Kind{Type: &v1alpha1.HttpsCertIssuer{}}, &handler.EnqueueRequestsFromMapFunc{
```
Comment (davidqhr) [inline controller/controllers/kalm_ns_controller.go:66]: I suggest make a func that returns a source.Kind pointer. Params is type and map function

Comment (mingmin-ddex) [reply controller/controllers/kalm_ns_controller.go]: updated

File: controller/controllers/storageclass_controller.go [hunk 2]
```diff
@@ -0,0 +1,243 @@
...
+
+	return "", false
+}
+
+func (r *StorageClassReconciler) reconcileDefaultStorageClass(cloudProvider string) error {
+	var expectedStorageClasses []v1.StorageClass
+
+	reclaimPolicy := corev1.PersistentVolumeReclaimRetain
+	switch cloudProvider {
+	//todo case "minikube":
```
Comment (davidqhr) [inline controller/controllers/storageclass_controller.go:72]: useless comment

File: controller/controllers/kalm_pv_controller.go [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+			return item, true
+		}
+	}
+
+	return
+}
+
+func (r *KalmPVReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	return ctrl.NewControllerManagedBy(mgr).
+		For(&corev1.PersistentVolume{}).
```
Comment (davidqhr) [inline controller/controllers/kalm_pv_controller.go:97]: Watch pvc? It seems this controller queries pvc.

Comment (mingmin-ddex) [reply controller/controllers/kalm_pv_controller.go:97]: this controller's task is to ensure all Kalm PVs are labeled with `kalm-pv`, this is done when PV is created, and PVC change won't affect this, so no need to watch PVC here.

File: controller/controllers/storageclass_controller.go (hunk 2 above)
Comment (mingmin-ddex) [reply controller/controllers/storageclass_controller.go]: updated

**Pull request merged**